### PR TITLE
feat: improve types generation

### PR DESCRIPTION
**Pull request by Tomas2D**
- Update dependencies 
- Because of updates, handle dates extraction
- Add support for handling `-` (minus character) in path name

### CODE REVIEW

File: scripts/types_generator/schema_transformer.py [hunk 1]
```diff
@@ -65,14 +66,29 @@ def _remove_compositions(schema: Any, *, path="", delimiter="."):
...
+        return re.sub(
+            r"\{(.*?)}",
+            lambda m: from_camel_case_to_snake_case(m.group(1)),
+            input,
+            flags=re.MULTILINE,
+        )
+
+    def _process_hyphens(input: str):
+        return re.sub(
+            pattern=r"[-](.)", repl=lambda x: f"_{x.group(1).upper()}", string=input, count=0, flags=re.MULTILINE
```
Comment (jezekra1) [inline scripts/types_generator/schema_transformer.py:88]: nice, I didn't know it was possible to access the group here!

File: scripts/types_generator/schema_transformer.py [hunk 2]
```diff
@@ -65,14 +66,29 @@ def _remove_compositions(schema: Any, *, path="", delimiter="."):
...
+
+    def _process_parameters(input: str):
+        return re.sub(
+            r"\{(.*?)}",
+            lambda m: from_camel_case_to_snake_case(m.group(1)),
+            input,
+            flags=re.MULTILINE,
+        )
+
+    def _process_hyphens(input: str):
```
Comment (jezekra1) [inline scripts/types_generator/schema_transformer.py:86]: Maybe we should stop calling things `input` when it's not necessary - it shadows the builtin `input` function.

It makes sense in the user-facing parameters, but here I'd call it `value` or something else :D

File: scripts/types_generator/extractor.py [hunk 3]
```diff
@@ -27,12 +29,13 @@ def extract_endpoints(api: Schema):
...
                 assert isinstance(value, dict)
                 if value.get("in") == "query" and value.get("name") == "version":
+                    version: datetime.date = value.get("schema", {}).get("enum")[0]
                     endpoints.append(
                         ApiEndpoint(
                             class_name=to_classname(f"{path_formatted}_{http_method.title()}_endpoint", True),
                             path=path,
                             method=str(http_method_raw).upper(),
-                            version=value.get("schema", {}).get("enum")[0],
+                            version=version.isoformat(),
```
Comment (jezekra1) [inline scripts/types_generator/extractor.py:38]: This breaks when I call `poe schema` on prod

Comment (Tomas2D) [reply scripts/types_generator/extractor.py:38]: You need to update your packages first.

Comment (jezekra1) [inline scripts/types_generator/extractor.py:38]: This breaks when I call `poe schema` against prod. Did you forget to parse it to date on line 32?

```
[2024-02-07 10:23:55][INFO][__main__][run]: Saving extracted endpoints information to '/Users/radek/GIT/ibm-generative-ai/src/genai/schema/_endpoints.py'
[2024-02-07 10:23:55][ERROR][__main__][run]: 'str' object has no attribute 'isoformat'
Traceback (most recent call last):
  File "/Users/radek/GIT/ibm-generative-ai/scripts/types_generator/main.py", line 99, in run
    for model in extract_endpoints(schema)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/radek/GIT/ibm-generative-ai/scripts/types_generator/extractor.py", line 38, in extract_endpoints
    version=version.isoformat(),
            ^^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'isoformat'
```

**Review by jezekra1 — CHANGES_REQUESTED** [review]

File: scripts/types_generator/extractor.py (hunk 3 above)
Comment (jezekra1) [reply scripts/types_generator/extractor.py:38]: it works after the update!

**Review by jezekra1 — APPROVED** [review]

**Pull request merged**